build.rs

```diff
@@ -39,15 +49,17 @@ fn main() {
     // FIXME: this exists to work around https://github.com/rust-lang/cargo/issues/10527
     println!("cargo:rerun-if-changed=crates/");
 
-    #[cfg(all(target_os = "linux", target_arch = "x86_64",))]
-    println!("cargo:rustc-cfg=enarx_with_shim");
+    if *IS_X86_64_LINUX {
```

Comment:
Is a `Lazy` necessary here? Furthermore, are two separate `if`s necessary here? How about a single `if` with:
```rust
if "linux" == var("CARGO_CFG_TARGET_OS").expect("missing CARGO_CFG_TARGET_OS")
    && "x86_64" == var("CARGO_CFG_TARGET_ARCH").expect("missing CARGO_CFG_TARGET_ARCH") {
```

Reply:
That's how existing code was organized and I did not change that. This organization also makes sense, because these are two distinctively different features we're enabling the way I see it.

IMO the "single `if`" approach is terrible for readability